cmd/crank/beta/trace/trace_test.go

```diff
@@ -0,0 +1,98 @@
+package trace
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestCmd_getResourceAndName(t *testing.T) {
+	type args struct {
+		Resource string
+		Name     string
+	}
+	type want struct {
+		resource string
+		name     string
+		err      error
+	}
+	tests := map[string]struct {
+		reason string
+		fields args
+		want   want
+	}{
+		"Splitted": {
+			reason: "Should return the resource and name if both are provided",
+			fields: args{
+				Resource: "resource",
+				Name:     "name",
+			},
+			want: want{
+				resource: "resource",
+				name:     "name",
+				err:      nil,
+			},
+		},
+		"OnlyResource": {
+			reason: "Should return an error if only resource is provided",
+			fields: args{
+				Resource: "resource",
+				Name:     "",
+			},
+			want: want{
+				err: cmpopts.AnyError,
+			},
+		},
+		"Empty": {
+			reason: "Should return an error if no resource is provided",
+			fields: args{
+				Resource: "",
+				Name:     "",
+			},
+			want: want{
+				err: cmpopts.AnyError,
+			},
+		},
+		"Combined": {
```

Comment:
would it be useful to have another test case that tests resource being specified as `resource/name` and `name` being specified as well? I think that may be the case @turkenh was mentioning earlier.

Reply:
Yes, same thing, I will add that

Reply:
done: https://github.com/crossplane/crossplane/pull/4910/commits/e0622f4d4d213291f4d28775dbc3c4c985b236d6